src/models/state/mod.rs

```diff
@@ -709,16 +811,34 @@ impl GlobalState {
 
             // store updated membership proof
             monitored_utxo.add_membership_proof_for_tip(tip_hash, membership_proof);
-            monitored_utxos.set(i, monitored_utxo);
+            setter.set(monitored_utxo);
         }
+        drop(iter_mut_lock); // <---- releases write lock.
 
         // Update sync label and persist
-        self.wallet_state
-            .wallet_db
-            .lock()
-            .await
-            .set_sync_label(tip_hash);
-        self.wallet_state.wallet_db.lock().await.persist();
+        self.wallet_state.wallet_db.set_sync_label(tip_hash);
+        self.wallet_state.wallet_db.persist();
+
+        Ok(())
+    }
+
+    pub async fn flush_databases(&mut self) -> Result<()> {
+        // flush wallet databases
+        self.wallet_state.wallet_db.persist();
+
+        let hash = self.chain.archival_state().get_latest_block().await.hash;
+
+        // persist archival_mutator_set, with sync label
+        self.chain
+            .archival_state_mut()
+            .archival_mutator_set
+            .set_sync_label(hash);
+        self.chain
+            .archival_state_mut()
+            .archival_mutator_set
+            .persist();
+
+        debug!("Persisted all databases");
 
         Ok(())
```

Comment:
It seems to me that there are some databases that are not being persisted here: `block_index_db` in `ArchivalState` and `peer_databases` in `NetworkingState`. Shouldn't they be flushed here as well?

Reply:
writing down what was discussed in conversation:

yeah, the flush calls are not necessary since `RustyLevelDB::batch_write()` sets the sync flag to true when calling `twenty_first::storage::leveldb::DB::write()`.    LevelDB guarantees that the data has been fully written to disk (synced/flushed) when the write call returns.

In fact, the C++ levelDB API does not offer a flush() method at all.   For that reason, I had removed it from RustyLevelDB.

I see though that you kept RustyLevelDB::flush() and modified it to perform an extra, empty write.

afaict, this extra write is un-necessary disk I/O.

That said, I think we are not calling flush() so frequently, so it probably won't be an issue for performance for the foreseeable future.    So I have added the flush calls back into this PR.

I also updated RustyLevelDB so that it defaults the sync flag to true, in case someone calls a setter other than batch_write() ie (put, delete).   Searching the code we don't seem to use these presently.  It might be best just to remove them RustyLevelDB, so that any future devs are not tempted to call them, and bypass batch writes.